core/src/main/java/org/elasticsearch/common/http/client/HttpDownloadHelper.java

```diff
@@ -153,7 +150,7 @@ public boolean downloadAndVerifyChecksum(URL checksumURL, Path originalFile, Pat
             // checksum file doesn't exist
             return false;
         } catch (IOException e) {
-            if (ExceptionsHelper.unwrapCause(e) instanceof FileNotFoundException) {
```

Comment:
you can remove the entire `catch (IOException e)` block here it's already handled correctly above